### DISCUSSION TITLE
[Detector Support]: Help With Stationary Car Detection

### DISCUSSION THREAD

**Discussion by moogoos** [blakeblackshear/frigate#15044] [Detector Support] [answered]
### Describe the problem you are having

On a street camera, the detection keeps detecting stationary cars parked up the road. I need some help tuning the config so that this is ignored. I dont well understand threshold and interval here to make an informed change and saw also the following confg in another discussion, but I'd like to try and understand these options. 

https://docs.frigate.video/configuration/reference/

I saw this in another post:

```
    detect:
      enabled: true
      width: 1280
      height: 720
      fps: 5
      stationary:
        max_frames:
          objects:
            car: 100
```

### Version

15-beta1

### Frigate config file

```yaml
objects:
      track:
        - person
        - car
        - truck
        - bicycle
        - motorcycle
        - dog
        - cat
        - mouse
    detect:
      enabled: true
      width: 1280
      height: 720
      fps: 5
```

### docker-compose file or Docker CLI command

```yaml
NA
```

### Relevant Frigate log output

```shell
NA
```

### Install method

Docker Compose

### Object Detector

OpenVino

### Screenshots of the Frigate UI's System metrics pages

NA

**Comment by NickM-27** [comment]
Can you describe what you mean by detecting stationary cars? In general this is Frigate working as intended, we'd need to understand what the specific issue is

https://docs.frigate.video/configuration/stationary_objects#why-does-frigate-track-stationary-objects

**Reply by moogoos** [reply]
ahh yeah that makes sense. Objects aren't necessarily moving. I guess what my issue is, is that if there is a car or two parked and every 5-10 minutes it's detected, which I guess it should be, and then records it as an event so i'll have 30 or so events that show as pictures in the event timeline, so it makes it difficult to distinguish between a more dynamic detection of an object, vs a static one that is repeated over and over.

![Screenshot 2024-11-18 at 10 51 52 AM](https://github.com/user-attachments/assets/f63c7c08-a2ab-4cc7-a9d3-bb5e49a7a923)

**Reply by moogoos** [reply]
I guess I could block the side of the street too to mitigate it some but trying to avoid that.

**Reply by NickM-27** [reply]
In 0.14 you'll have to see if you can catch that happening in the debug live view. Detections will only show when an object is active, no stationary. This could be due to cars driving by and occluding the car or perhaps due to motion being detected and the bounding boxes not being consistent

**Reply by moogoos** [reply]
Hello, Are you saying ill have to downgrade to .14 to troubleshoot? This is causing a constant "alert" for some reason in my history. This camera is constantly in red frame:

![Screenshot 2024-11-20 at 3 12 22 PM](https://github.com/user-attachments/assets/43b56de8-6a71-4662-8097-3c09d2e38bcc)

**Reply by moogoos** [reply]
![Screenshot 2024-11-20 at 4 46 12 PM](https://github.com/user-attachments/assets/3dd95168-e705-4799-bcd7-d47e48734ea7)

**Reply by moogoos** [reply]
"All person, car objects on Trellis will be shown as Alerts."

**Reply by moogoos** [reply]
![Screenshot 2024-11-20 at 4 49 50 PM](https://github.com/user-attachments/assets/dd8f0261-2b8e-4c4d-bdcb-51ec8ce6675c)

**Reply by moogoos** [reply]
![Screenshot 2024-11-20 at 4 51 28 PM](https://github.com/user-attachments/assets/1f298b3c-550f-47af-87d3-7be0c79fe231)

**Reply by NickM-27** [reply]
No, if you're running 0.15 you can just use the object lifecycle view

I'd suggest using a zone and make it required for alerts so only areas you care about are alerts

**Reply by moogoos** [reply]
alright ill give it a go. I already have the road and sidewalk as zones so will see if that helps

**Reply by moogoos** [reply]
adding zones didn't work unfortunately. This is so odd. This camera feed is constantly red framed detecting these stationary cars. I even tried blocking it out but it still shows the bounding boxes:

![Screenshot 2024-11-21 at 3 00 18 PM](https://github.com/user-attachments/assets/7664699b-991b-4102-8bee-2d50fff6a1d5)

**Reply by moogoos** [reply]
![Screenshot 2024-11-21 at 3 01 10 PM](https://github.com/user-attachments/assets/ddcb3c1a-f653-422c-b907-faa3464cff67)

**Reply by moogoos** [reply]
i assume since the bounding boxes are still bouncing around is what causes the frame to stay in constant alert?

**Reply by NickM-27** [reply]
> i assume since the bounding boxes are still bouncing around is what causes the frame to stay in constant alert?

yes, this is a difficult scenario because they are small, scoring low, and overlapping. You can configure a minimum area for cars and that would stop it, but if that zone is required then they should not be created alert review items anymore

**Reply by moogoos** [reply]
whats interesting is they aren't in a zone at all anymore or shouldn't be. Ive forced it to the "road" zone only, which is the brown outline of the road, and masked the cars that are no longer in a zone as well but they are getting included.

 I'm ok with a minumum area. I dont need that far up really. How do i do that min area for an object?

**Reply by NickM-27** [reply]
one of them looks really close, remember that zones are determined by the bottom center of the bounding box

**Reply by moogoos** [reply]
I tried a less aggressive zone that is far away from the car bounding boxes and it still is red framed.

![Screenshot 2024-11-21 at 3 54 18 PM](https://github.com/user-attachments/assets/600bd3d6-7417-4281-a7fe-f41f03bcd0bc)


How do I adjust the min area for an object?

**Reply by moogoos** [reply]
![Screenshot 2024-11-21 at 4 11 49 PM](https://github.com/user-attachments/assets/84c8f64e-007b-400e-a908-976af6917dae)

**Reply by NickM-27** [reply]
Zones won't affect the red outline, that will always show for any objects that are detected as active

**Comment by zac9** [comment, marked as answer]
You can filter you cars based on minimum size. This is an example from my system to filter people and cars in the street. I have it attached to my cam config. 

```
 objects:
      track:
        - person
        - animal
      filters:
        person:
          min_area: 2000
        car:
          min_area: 10000
```

**Reply by moogoos** [reply]
cool thanks will give this a go

**Comment by moogoos** [comment]
using min_area was the most effective way to deal with this position. Thank you both @zac9  and @NickM-27 for the support
